src/Avalonia.Controls/TreeViewItem.cs

```diff
@@ -65,6 +75,34 @@ public bool IsSelected
             set { SetValue(IsSelectedProperty, value); }
         }
 
+        /// <summary>
+        /// Gets or sets the depth of the item.
+        /// </summary>
+        public int Depth
+        {
+            get { return this.GetDepth(this); }
```

Comment:
Isn't it possible to do this in the ContainerGenerator? Caching it would also be nice.

Reply:
@Gillibald I have a look into this, need to do a little housekeeping on the PR in any case.

Reply:
Yes this could be calculated when added to the logical tree 👍

Reply:
That's exactly where it lives now, essentially "cached" too...